addons/crm/models/crm_team.py

```diff
@@ -47,16 +47,10 @@ def _compute_lead_unassigned_count(self):
         for team in self:
             team.lead_unassigned_count = counts.get(team.id, 0)
 
+    @api.depends('crm_team_member_ids.lead_month_count')
     def _compute_lead_all_assigned_month_count(self):
-        limit_date = datetime.datetime.now() - datetime.timedelta(days=30)
-        leads_data = self.env['crm.lead'].read_group([
-            ('team_id', 'in', self.ids),
-            ('date_open', '>=', fields.Datetime.to_string(limit_date)),
-            ('user_id', '!=', False),
-        ], ['team_id'], ['team_id'])
-        counts = {datum['team_id'][0]: datum['team_id_count'] for datum in leads_data}
         for team in self:
-            team.lead_all_assigned_month_count = counts.get(team.id, 0)
+            team.lead_all_assigned_month_count = sum(s.lead_month_count for s in team.crm_team_member_ids)
```

Comment:
Difference is
  * before: count of leads in team, with an user
  * after: count of leads in team with an user member of the team

I changed because in views, you could have sum(member leads) < team leads which make no sense. Having lead assigned to an user being in another team should not happen, and if it happens those are not counted in assignement information.